docs/development.md

```diff
@@ -0,0 +1,162 @@
+
+If you have a large machine, you might want to increase the `--max-concurrency` beyond 1. One of the benefits of hosted infrastructure (non-local) is that extremely high concurrency is possible.
+
+Metadata about the attemts, errors, build logs, and the rebuilt artifacts will all be written to the local asset store (/tmp/oss-rebuild).
+
+For example, if you executed the above benchmark exactly once, you would have:
+
+```
+$ tree /tmp/oss-rebuild
+/tmp/oss-rebuild
+├── assets
+│   └── pypi
+│       └── absl_py
+│           └── 2.0.0
+│               └── absl_py-2.0.0-py3-none-any.whl
+│                   └── 2025-12-01T23:46:16Z
+│                       ├── absl_py-2.0.0-py3-none-any.whl
+│                       └── logs
+└── rundex
+    ├── runs
+    │   └── 2025-12-01T23:46:16Z
+    │       └── pypi
+    │           └── absl_py
+    │               └── absl_py-2.0.0-py3-none-any.whl
+    │                   └── firestore.json
+    └── runs_metadata
+        └── 2025-12-01T23:46:16Z.json
+14 directories, 4 files
+```
+
+You can see metadata about the run as a whole is stored under /tmp/oss-rebuild/rundex/runs_metadata/2025-12-01T23:46:16Z.json
+
+Target-specific metadata can be found under /tmp/oss-rebuild/rundex/runs/...
+
+Finally, all the logs, rebuilt artifacts, and other assets will be located under /tmp/oss-rebuild/assets/...
+
+For more details about where OSS Rebuild stores artifacts, see <TODO: storage.md>
+
+## `get-results` - Querying Results
+
+Once a local run is complete, you can query the outcomes programmatically using the get-results command. You'll need the run id, which logged at the start of `run-bench`. In the above example it was `2025-12-01T23:46:16Z`.
+
+To query the results, run `get-results` and provide the run id. For example:
+
+```bash
+go run ./tools/ctl get-results --run <run-id>
+```
+
+By default, `get-results` will summarize the run's results, grouping targets by their finally result message. Alternatively, you can format the output as a csv for analysis or visualization in your favorite spreadsheet editor:
+
+```bash
+go run ./tools/ctl get-results --run <run-id> --format=csv
+```
+
+If you want to filter the results based on their message, you can use the `--pattern` flag, providing a regex pattern to match results that are included in the output. These are the same messages used for grouping in the default `get-results` query. For example, you can select all targets that finished building but did not match the upstream:
+
+```bash
+go run ./tools/ctl get-results --run <run-id> --pattern='.*rebuild content mismatch.*' --format=csv
+```
+
+Additionally, by combining filtering with `--format=bench` it's possible to make a new benchmark with only the failure case we're interested in. This is helpful if you fix a specific failure case and want to retry only those errors.
```

Comment:
"we're": might want to go through the text and confirm the subject you're using. I think "you" gets used here, as well.